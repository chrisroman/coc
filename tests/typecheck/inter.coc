let A0 = [t: *][x: t]t in
(\a: [_1: [_2: A0]*]*)(\x: A0)[P : [_3 : A0]*][_4 : (a P)](P x)
