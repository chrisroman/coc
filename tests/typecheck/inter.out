let A0 = [t: *] [x: t] t in 
(λa: [_1: [_2: A0] *] *) (λx: A0) [P: [_3: A0] *] [_4: (a P)] (P x)
{A0} typecheck_single_term [t: *] [x: t] t
Type: *
typecheck_single_term (λa: [_1: [_2: [t: *] [x: t] t] *] *) (λx: [t: *] [x: t] t) [P: [_3: [t: *] [x: t] t] *] [_4: (a P)] (P x)
Type: [a: [_1: [_2: [t: *] [x: t] t] *] *] [x: [t: *] [x: t] t] *
Value: (λa: [_1: [_2: [t: *] [x: t] t] *] *) (λx: [t: *] [x: t] t) [P: [_3: [t: *] [x: t] t] *] [_4: (a P)] (P x)
